functional/FunctionalTest.java

```diff
@@ -4242,6 +4249,308 @@ public static void selectObjectContent_test1() throws Exception {
     }
   }
 
+  /** Test: setBucketTags(SetBucketTagsArgs args). */
+  public static void setBucketTags_test() throws Exception {
+    if (!mintEnv) {
+      System.out.println("Test: setBucketTags(SetBucketTagsArgs args)");
+    }
+
+    long startTime = System.currentTimeMillis();
+    String bucketName = getRandomName();
+    try {
+      client.makeBucket(MakeBucketArgs.builder().bucket(bucketName).build());
+      try {
+        Map<String, String> map = new HashMap<>();
+        map.put("Project", "Project One");
+        map.put("User", "jsmith");
+        client.setBucketTags(SetBucketTagsArgs.builder().bucket(bucketName).tags(map).build());
+        mintSuccessLog("setBucketTags(SetBucketTagsArgs args)", null, startTime);
+      } finally {
+        client.removeBucket(RemoveBucketArgs.builder().bucket(bucketName).build());
+      }
+    } catch (Exception e) {
+      ErrorResponse errorResponse = null;
+      if (e instanceof ErrorResponseException) {
+        ErrorResponseException exp = (ErrorResponseException) e;
+        errorResponse = exp.errorResponse();
+      }
+
+      // Ignore NotImplemented error
+      if (errorResponse != null && errorResponse.errorCode() == ErrorCode.NOT_IMPLEMENTED) {
+        mintIgnoredLog("setBucketTags(SetBucketTagsArgs args)", null, startTime);
+      } else {
+        mintFailedLog(
+            "setBucketTags(SetBucketTagsArgs args)",
+            null,
+            startTime,
+            null,
+            e.toString() + " >>> " + Arrays.toString(e.getStackTrace()));
+        throw e;
+      }
+    }
+  }
```

Comment:
1. I avoided that for repetition of code
```java
    } catch (ErrorResponseException e) {
      // Ignore NotImplemented error
      if (e.errorResponse().errorCode() == ErrorCode.NOT_IMPLEMENTED) {
        mintIgnoredLog("deleteObjectTags(String bucketName)", null, startTime);
      } else {
        mintFailedLog(
            "deleteObjectTags(String bucketName)",
            null,
            startTime,
            null,
            e.toString() + " >>> " + Arrays.toString(e.getStackTrace()));
        throw e;
      }
    } catch (Exception e) {
        mintFailedLog(
            "deleteObjectTags(String bucketName)",
            null,
            startTime,
            null,
            e.toString() + " >>> " + Arrays.toString(e.getStackTrace()));
        throw e;
    }
```
2. This problem is there across all tests. We would need to find a better way to do that and fix everywhere. I am limiting the scope of this PR.

Reply:
1. In that case, I would suggest simplifying the code as follows:

```java
    } catch (Exception e) {
      if (e instanceof ErrorResponseException) {
        if (((ErrorResponseException) e).errorResponse().errorCode() == ErrorCode.NOT_IMPLEMENTED) {
          mintIgnoredLog("setBucketTags(SetBucketTagsArgs args)", null, startTime);
          return;
        }
      }
      mintFailedLog(
          "setBucketTags(SetBucketTagsArgs args)",
          null,
          startTime,
          null,
          e.toString() + " >>> " + Arrays.toString(e.getStackTrace()));
      throw e;
    }
```

2. Since this code is repeated many times in this PR itself, I suggest to create a method and use it only in these places as of now, thus not increasing the scope of the PR. Using the same (or a similar) method in all other places can be done in a separate PR.

Reply:
done